src/types/boolean.md

```diff
@@ -1,17 +1,52 @@
 # Boolean type
 
-The `bool` type is a datatype which can be either `true` or `false`. The boolean
-type uses one byte of memory. It is used in comparisons and bitwise operations
-like `&`, `|`, and `!`.
-
 ```rust
-fn main() {
-    let x = true;
-    let y: bool = false; // with the boolean type annotation
-
-    // Use of booleans in conditional expressions
-    if x {
-        println!("x is true");
-    }
-}
+let b: bool = true;
 ```
+
+The *boolean type* or *bool* is a primitive data type that can take on one of
+two values, called *true* and *false*.
+
+Values of this type may be created using a [literal expression] using the
+keywords `true` and `false` corresponding to the value of the same name.
+
+This type is a part of the [language prelude] with the [name] `bool`.
+
+An object with the boolean type has a [size and alignment] of 1 each. The
+value false has the bit pattern `0x00` and the value true has the bit pattern
+`0x01`. It is [undefined behavior] for an object with the boolean type to have
+any other bit pattern.
+
+The boolean type is the type of many operands in various [expressions]:
+
+* The condition operand in [if expressions] and [while expressions]
+* The operands in [lazy boolean operator expressions][lazy]
+
+> **Note**: The boolean type acts similarly to but is not an [enumerated type].
+In practice, this mostly means that constructors are not associated to the type
+(e.g. `bool::true`).
+
+Like all primitives, the boolean type [implements][p-impl] the
```

Comment:
Is there a reason this doesn't list other traits like the [operator traits](https://doc.rust-lang.org/nightly/reference/special-types-and-traits.html#operator-traits)?  It does seem like a little bit of a gray area of which trait impls are part of the "language", but it is not clear to me how that line is drawn.

Reply:
I just referenced the traits on the special traits page, and didn't think too much about the operator traits. It is an oversight, sorta. There's inherent behavior with the operators themselves, but the trait implementations are [library-side](https://doc.rust-lang.org/src/core/ops/bit.rs.html#167):

```rust
macro_rules! bitand_impl {
    ($($t:ty)*) => ($(
        #[stable(feature = "rust1", since = "1.0.0")]
        impl BitAnd for $t {
            type Output = $t;

            #[inline]
            fn bitand(self, rhs: $t) -> $t { self & rhs }
        }

        forward_ref_binop! { impl BitAnd, bitand for $t, $t }
    )*)
}

bitand_impl! { bool usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
```

Comparison operators are also built-in that way.

---

How about I add truth tables for all the things you can do with a boolean, and then link to those from the operator expressions?